src/canmatrix/canmatrix.py

```diff
@@ -815,67 +815,55 @@ def calcDLC(self):
-        Numbering starts from one.
+        Represents the bit usage in the frame by means of a list with n items (n bits of frame length).
+        Every item represents one bit and contains a list of signals (object refs) with each signal, occupying that bit.
+        Bits with empty list are unused.
 
-        Bits with "zero" index are unused.
-
-        Example: [2, 2, 2, 1, 1, 0, 0, 3, 3, 3, 3, 0, 0, ...]
-
-        :return: list with signal "index plus one" in every bit. Zeros mean 'unused'.
-        :rtype: list of int
+        Example: [[], [], [], [sig1], [sig1], [sig1, sig5], [sig2, sig5], [sig2], []]
+        :return: list of lists with signalnames
+        :rtype: list of lists
         """
-        bitfield = []
-        bitfieldLe = []
-        bitfieldBe = []
-
-        for i in range(0,64):
-            bitfieldBe.append(0)
-            bitfieldLe.append(0)
-            bitfield.append(0)
-        i = 0
-
-        for sig in self.signals:
-            i += 1
-            for bit in range(sig.getStartbit(),  sig.getStartbit() + int(sig.size)):
-                if sig.is_little_endian:
-                    bitfieldLe[bit] = i
-                else:
-                    bitfieldBe[bit] = i
-
-        for i in range(0,8):
-            for j in range(0,8):
-                bitfield[i*8+j] = bitfieldLe[i*8+(7-j)]
+        little_bits = [[] for _dummy in range(0, (self.size * 8))]
+        big_bits = [[] for _dummy in range(0, (self.size * 8))]
+        for signal in self.signals:
+            if signal.is_little_endian:
+                least = self.size * 8 - signal.startBit
+                most = least - signal.size
+                for i in range(most, least):
+                    little_bits[i].append(signal)
+            else:
+                most = signal.startBit
+                least = most + signal.size
+                for i in range(most, least):
+                    little_bits[i].append(signal)
 
-        for i in range(0,8):
-            for j in range(0,8):
-                if bitfield[i*8+j] == 0:
-                    bitfield[i*8+j] = bitfieldBe[i*8+j]
+        little_bits = reversed(tuple(grouper(little_bits, 8)))
+        little_bits = tuple(chain(*little_bits))
 
+        returnList = list()
+        for i in range(0,self.size * 8):
+            returnList.append(little_bits[i] + big_bits[i])
 
-        return bitfield
+        return returnList
 
-    def createDummySignals(self):
+    def create_dummy_signals(self):
         """Create big-endian dummy signals for unused bits.
 
         Names of dummy signals are *_Dummy_<frame.name>_<index>*
         """
-        bitfield = self.findNotUsedBits()
-        # for i in range(0,8):
-        #    print (bitfield[(i)*8:(i+1)*8])
+        bitfield = self.get_frame_layout()
         startBit = -1
         sigCount = 0
-        for i in range(0,64):
-            if bitfield[i] == 0 and startBit == -1:
+        for i in range(0,self.size * 8):
```

Comment:
```suggestion
        for i in range(self.size * 8):
```

How about using `len(bitfield)` throughout instead of `self.size * 8`?

And could we do something like below?  We do use the index but at least we can drop the `range()` and indexing

```python3
for index, bit_signals in enumerate(bitfield):
```

Reply:
@ebroecker, while we do need the index in this case it's nice to at least tie the iteration to some existing iterable.  So this is about `for index, bit_signals in enumerate(bitfield):` and also using `len(bitfield)` everywhere in this function where there is presently `self.size * 8`.

Reply:
OK, usage of len(bitfield) is nicer.
The other idea - using  `enumerate(bitfield)` - is this more easy to understand than using the index? If so, I'll change

Reply:
Most of the time we don't care about indexes so using
```python
for i in range(len(things)):
    do_something_with(things[i])
```
is much more simply written as
```
for thing in things:
    do_something_with(thing)
```
On the occasion that we do care about indexes that is easily expanded to
```
for i, thing in enumerate(things):
    do_something_with(i, thing)
```
which still leaves us iterating over the stuff of interest but also getting an index for each item.  I suppose there's room for opinion here but I much prefer using `enumerate()` to add an index to iteration than to not even iterate over the thing i want each element of.

Reply:
Ok, I think I understand.
But this time, we only need the index and do not need the content of bitfield. 
So this time it seems to me more confusing to me to iterate while I only need the index, or do I miss something?

Reply:
Aren't there at least a couple accesses to `bitfield[i]`?

![image](https://user-images.githubusercontent.com/543719/49311082-18a8c280-f4ae-11e8-9139-91f5e2ed45c8.png)

Reply:
OK, got it, convinced.